src/main/java/dev/ecattez/shahmat/board/ImpossibleToMove.java

```diff
@@ -0,0 +1,9 @@
+package dev.ecattez.shahmat.board;
+
+public class ImpossibleToMove extends RulesViolation {
+
+    public ImpossibleToMove(Piece piece, Square from, Square to) {
```

Comment:
All of the `toString`s are implemented ?

Reply:
There are exceptions, no need to implements this method.